### PR TITLE
Coding style adjustments

### PR DESCRIPTION
**Pull request by beru**
Hi, this PR adjusts coding style.

### CODE REVIEW

File: tiny_dnn/layers/max_unpooling_layer.h [hunk 1]
```diff
@@ -100,20 +100,20 @@ class max_unpooling_layer : public feedforward_layer<Activation> {
         for_(parallelize_, 0, in2out_.size(), [&](const blocked_range& r) {
             for (int i = r.begin(); i < r.end(); i++) {
                 const auto& in_index = out2in_[i];
-                a[i] = (max_idx[in_index] == i) ? in[in_index] : float_t(0);
+                a[i] = (max_idx[in_index] == i) ? in[in_index] : float_t{0}3;
```
Comment (Randl) [inline tiny_dnn/layers/max_unpooling_layer.h]: typo

**Comment by Randl** [comment]
Is there any principal difference between `float_t{0}` and `float_t(0)`?

File: tiny_dnn/layers/max_unpooling_layer.h (hunk 1 above)
Comment (beru) [inline tiny_dnn/layers/max_unpooling_layer.h]: @Randl Thank you for the notification.

**Comment by beru** [comment]
> Is there any principal difference between float_t{0} and float_t(0)?

Since both of them hande compile time constant `0`, I guess there wouldn't be any practical difference. It can be compiler-specific after all, I just removed some C-style casts code in the library in the name of brainwashing. 

`float_t{0}` this one is probablly called C++11's uniform initialization.
`(float)0` and this one is mighty C-style cast and some people dislike this for several reasons
`float_t(0)` It looks like the last one is called direct-initialization.

I personally prefer the one in the middle but...

**Comment by edgarriba** [comment]
@beru thanks for this PR. However, how do you think that we can maintain this code style? Should we provide tools for code formatting such git hooks? we discussed this idea some time ago

> float_t{0} this one is probablly called C++11's uniform initialization.

We are "selling" the library as pure C++11 so you answered yourself :smile:

**Comment by beru** [comment]
> how do you think that we can maintain this code style? Should we provide tools for code formatting such git hooks? we discussed this idea some time ago

Some form of consensus might be needed before adopting formatting guidelines.
I'll list candidates of code formats afrer eons of time.

I can say 80 characters per line rule is pretty harsh since there are a lot files that don't follow it.

By the way, I forgot to check the result of `$ bash -x .travis/cpplint_test.sh`, so I'll do it.
Talking about other statick checkers, I've used `Cppcheck` in the past. I remember it was pretty good.
And I still haven't used [Vera++](https://github.com/verateam/vera).

**Comment by beru** [comment]
>> float_t{0} this one is probablly called C++11's uniform initialization.

> We are "selling" the library as pure C++11 so you answered yourself 😄

Ont thing I noticed is uniform initialization cannot be used for narrowing conversions, so I have to use `static_cast` in that case.

**Comment by edgarriba** [comment]
@beru vera++ looks good. However, remember that there was also a PR that uses clang-format https://github.com/tiny-dnn/tiny-dnn/pull/363

I personally prefer to provide a git hook or something similar which can let the user be IDE independent.

**Comment by beru** [comment]
> However, remember that there was also a PR that uses clang-format #363

I've just checked it briefly. I guess a lot of people prefer auto-formatting to ease efforts to rewrite code.

By the way, right now, 4 spaces indent is used. Have you decided to switch to 2 spaces? And what about 80 characters per line rule? I personally prefer 4 spaces with **NO** 80 characters per line rule. I guess setting rules and making decisions with endless communication take a lot of efforts and time. :sweat: So let's continue playing a waiting game.

**Comment by beru** [comment]
@edgarriba Talking about clang-format, I've just found an interesting web site.
https://clangformat.com/

**Comment by beru** [comment]
> I personally prefer to provide a git hook or something similar which can let the user be IDE independent.

People should locally apply code formatting with whatever the tools they have. (auto/manual) That should give us more freedom. But I think precise code style guidelines should be given.

**Comment by edgarriba** [comment]
@beru I opened https://github.com/tiny-dnn/tiny-dnn/issues/495
let's decide there

**Comment by bhack** [comment]
https://github.com/mantidproject/mantid/wiki/clang-format

**Comment by Randl** [comment]
I'll try to make https://github.com/tiny-dnn/tiny-dnn/pull/363 working this week if nobody does it until then

**Comment by edgarriba** [comment]
@beru rebase so that I can merge

**Comment by beru** [comment]
@edgarriba Please merge this PR.

**Pull request merged**